lib/ansible/modules/cloud/azure/azure_rm_keyvault.py

```diff
@@ -0,0 +1,436 @@
+            state=dict(choices=['present', 'absent'], default='present', type='str'),
+            enabled_for_deployment=dict(default=False, type='bool'),
+            enabled_for_disk_encryption=dict(default=False, type='bool'),
+            enabled_for_template_deployment=dict(default=False, type='bool'),
+            location=dict(type='str'),
+            sku=dict(choices=['standard', 'premium'], default='standard', type='str'),
+            vault_uri=dict(type='str'),
+            tenant_id=dict(type='str'),
+            object_id=dict(type='str'),
+            application_id=dict(type='str'),
+            permissions=dict(choices=[
+                'Key, Secret, & Certificate Management',
+                'Key & Secret Management',
+                'Key Management',
+                'Secret Management',
+                'Certificate Management',
+                'SQL Server Connector',
+                'Azure Backup',
+                'Azure Data Lake Store'
+            ], default='Key, Secret, & Certificate Management', type='str')
+        )
+
+        # store the results of the module operation
+        self.results = dict(
+            changed=False,
+            state=dict()
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.state = None
+        self.enabled_for_deployment = None
+        self.enabled_for_disk_encryption = None
+        self.enabled_for_template_deployment = None
+        self.location = None
+        self.sku = None
+        self.vault_uri = None
+        self.tenant_id = None
+        self.object_id = None
+        self.application_id = None
+        self.permissions = None
+        self.tags = None
+
+        super(AzureRMKeyVault, self).__init__(self.module_arg_spec,
+                                              supports_check_mode=True,
+                                              supports_tags=True)
+
+    def exec_module(self, **kwargs):
+
+        # create a new vault variable in case the 'try' doesn't find a vault
+        vault = None
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        self.results['check_mode'] = self.check_mode
+
+        # retrieve resource group to make sure it exists
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            # Set default location
+            self.location = resource_group.location
+
+        changed = False
+        results = dict()
+
+        try:
+            self.log('Fetching Vault {0}'.format(self.name))
+            vault = self.keyvault_client.vaults.get(self.resource_group, self.name)
+
+            # serialize object into a dictionary
+            results = vault.as_dict()
+
+            # don't change anything if creating an existing vault (unless outdated tags), but change if deleting it
+            if self.state == 'present':
+                changed = False
```

Comment:
No update at all? Why not delete the old one and create a new one if some configuration changed?